src/java.base/share/classes/jdk/internal/classfile/impl/StackCounter.java

```diff
@@ -307,14 +303,10 @@ public StackCounter(LabelContext labelContext,
                     case INVOKEVIRTUAL, INVOKESPECIAL, INVOKESTATIC, INVOKEINTERFACE, INVOKEDYNAMIC -> {
                         var cpe = cp.entryByIndex(bcs.getIndexU2());
                         var nameAndType = opcode == INVOKEDYNAMIC ? ((DynamicConstantPoolEntry)cpe).nameAndType() : ((MemberRefEntry)cpe).nameAndType();
-                        var mDesc = MethodTypeDesc.ofDescriptor(nameAndType.type().stringValue());
-                        for (var arg : mDesc.parameterList()) {
-                            addStackSlot(-TypeKind.from(arg).slotSize());
-                        }
+                        addStackSlot(-countMethodStack(nameAndType.type(), true));
```

Comment:
Can use something like:
```java
var mtd = Util.methodTypeSymbol(nameAndType);
addStackSlot(Util.slotSize(mtd.returnType()) - Util.parameterSlots(mtd));
```
if you stick with MTD.

Reply:
Calculation of MTD for each method invocation is extremely ineffective way to just count the size.